tests/functional/test_multimodel.py

```diff
@@ -0,0 +1,167 @@
+"""Functional test for :func:`esmvalcore.preprocessor._multimodel`."""
+
+from pathlib import Path
+
+import iris
+import numpy as np
+import pytest
+
+from esmvalcore.preprocessor import multi_model_statistics
+
+esmvaltool_sample_data = pytest.importorskip("esmvaltool_sample_data")
+
+
+def preprocess_data(cubes):
+    first_cube = cubes[0]
+    t1, t2 = 0, 50  # time-slicing first N items
+
+    # regrid to first cube
+    regrid_kwargs = {
+        'grid': first_cube,
+        'scheme': iris.analysis.Linear(),
+    }
+
+    cubes = [cube[t1:t2] for cube in cubes]
+    cubes = [cube.regrid(**regrid_kwargs) for cube in cubes]
+
+    return cubes
+
+
+@pytest.fixture
+def timeseries_cubes_amon():
+    """Representative timeseries data."""
+    cubes = esmvaltool_sample_data.load_timeseries_cubes(
+        mip_table='Amon')
+    cubes = preprocess_data(cubes)
+    return cubes
+
+
+@pytest.fixture
+def timeseries_cubes_day():
+    """Representative timeseries data."""
+    cubes = esmvaltool_sample_data.load_timeseries_cubes(
+        mip_table='day')
+    cubes = preprocess_data(cubes)
+    return cubes
+
+
+def multimodel_test(cubes, span, statistic):
+    statistics = [statistic]
+
+    output = multi_model_statistics(cubes, span=span, statistics=statistics)
+
+    return output
+
+
+def multimodel_regression_test(name, cubes, span):
+    statistic = 'mean'
+    output = multimodel_test(cubes, span=span, statistic=statistic)
+
+    # this fails for span='overlap', for which `output == cubes` (???)
+    assert isinstance(output, dict)
+
+    output_cube = output[statistic]
+
+    # NOTE for the regression test
+    # The following test will fail if the data are changed or if the
+    # multimodel code changes significantly. To update the data for the
+    # regression test, remove the corresponding `.nc` files.
+    filename = Path(__file__).with_name(f'{name}-{span}-{statistic}.nc')
+    if filename.exists():
+        expected_cube = iris.load(str(filename))[0]
+        assert np.allclose(output_cube.data, expected_cube.data)
```

Comment:
It would be good to use this function
https://github.com/ESMValGroup/ESMValCore/blob/c07e280274ef00aba2ad12969e22ca43660d5d5d/tests/__init__.py#L10-L14
here so the mask is also checked

and maybe add some other checks too, e.g. on the metadata and coordinates